### PR TITLE
Add contour line plot [com1; ana5; out3; breaking change]

### PR DESCRIPTION
**Pull request by awirb**
* add a contour line plot for peak files 
* runScript example

Breaking change for override sections: now need the collection name as well as the module name

- [x] add info to documentation

### CODE REVIEW

File: avaframe/ana5Utils/distanceTimeAnalysis.py [hunk 1]
```diff
@@ -580,7 +581,14 @@ def initializeRangeTime(modName, cfg, dem, simHash):
     """
 
     # fetch configuration and add info
-    cfgRangeTime = cfgUtils.getModuleConfig(modName)
+    if cfg.has_section('distanceTimeAnalysis_override'):
```
Comment (fso42) [inline avaframe/ana5Utils/distanceTimeAnalysis.py:584]: mh, I guess this is up for discussion, but should we really make these special treatments? Or would it be possible to use the getModuleConfig instead?

File: avaframe/com1DFA/com1DFACfg.ini [hunk 2]
```diff
@@ -446,3 +446,16 @@ unitpfv = ms-1
 # peak files and plots are exported, option to turn off exports when exportData is set to False
 # this affects export of peak files and also generation of peak file plots
 exportData = True
+
+
+[distanceTimeAnalysis_override]
```
Comment (fso42) [inline avaframe/com1DFA/com1DFACfg.ini:451]: Mh, I'm not sure this is a good idea. I think this module should be purely computation and only have a flag that produces data for another module, but not have override for that...

File: avaframe/out3Plot/outQuickPlot.py [hunk 3]
```diff
@@ -580,3 +580,64 @@ def plotContours(contourDict, resType, thresholdValue, pathDict, addLegend=True)
...
+        ------------
+        avaDir: str or pathlib path
+            path to avalanche directory
+        modName: str
+            name of computational module
+        resType: str
+            result variable type
+        level: float
+            contour line level
+        specDir: str or pathlib path
```
Comment (fso42) [inline avaframe/out3Plot/outQuickPlot.py:600]: minor: what does specDir mean? Why not peakFileDir or something more descriptive?

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com1DFA/com1DFACfg.ini (hunk 2 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFACfg.ini]: this is changed now - one function to create new local config files for override modules

File: avaframe/ana5Utils/distanceTimeAnalysis.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 
 # Local imports
 from avaframe.in3Utils import cfgUtils
+from avaframe.in3Utils import cfgHandling
```
Comment (fso42) [inline avaframe/ana5Utils/distanceTimeAnalysis.py:16]: What is this one needed for?

File: avaframe/com1DFA/com1DFACfg.ini [hunk 5]
```diff
@@ -417,7 +417,7 @@ contourResType = ppr
 # thresholdValue for contour lines plot
 thresholdValue = 1.
 # create data required for range-time diagram of simulation results -
-# additional configurations in ana5Utils - distanceTimeAnalysisCfg
+# additional configurations in section distanceTimeAnalysis_override
```
Comment (fso42) [inline avaframe/com1DFA/com1DFACfg.ini:420]: Should we point to a general information about override sections?

File: avaframe/in3Utils/cfgHandling.py [hunk 6]
```diff
@@ -497,3 +504,53 @@ def errorDuplicateListEntry(listKeys, message):
...
+
+         Parameters
+         -----------
+         cfgFull: configparser
+            configuration with override sections for modules
+
+    """
+
+    for section in cfgFull.sections():
+        if '_override' in section:
```
Comment (fso42) [inline avaframe/in3Utils/cfgHandling.py:524]: generate a list first of these section and then loop over that one -> to reduce the complexity of this code
This is hard to read...

File: avaframe/in3Utils/cfgUtils.py [hunk 7]
```diff
@@ -93,11 +94,15 @@ def getModuleConfig(module, fileOverride='', modInfo=False, toPrint=True, onlyDe
...
-    modName = str(pathlib.Path(module.__file__).stem)
+    if isinstance(module, pathlib.Path):
+        modPath = module.parent
+        # get filename of module
+        modName = module.stem
+    else:
+        # get path of module
+        modPath = pathlib.Path(module.__file__).resolve().parent
+        # get filename of module
+        modName = str(pathlib.Path(module.__file__).stem)
```
Comment (fso42) [inline avaframe/in3Utils/cfgUtils.py:105]: Since this is also in another function -> maybe make a callable function out of this...

**Review by fso42 — CHANGES_REQUESTED** [review]
I think it would be necessary to explain the override logic in our documentation...

File: avaframe/ana5Utils/distanceTimeAnalysis.py (hunk 1 above)
Comment (awirb) [reply avaframe/ana5Utils/distanceTimeAnalysis.py]: removed

File: avaframe/ana5Utils/distanceTimeAnalysis.py (hunk 4 above)
Comment (awirb) [reply avaframe/ana5Utils/distanceTimeAnalysis.py:16]: not needed anymore

File: avaframe/com1DFA/com1DFACfg.ini (hunk 5 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFACfg.ini:420]: now pointing again to ana5Utils/distanceTimeAnalysisCfg or its locl copy

File: avaframe/in3Utils/cfgUtils.py (hunk 7 above)
Comment (awirb) [reply avaframe/in3Utils/cfgUtils.py:105]: moved to a function

File: docs/configuration.rst [hunk 8]
```diff
@@ -53,8 +55,18 @@ In the configuration file itself, there are multiple options to vary a parameter
...
+An example of this usage can be found in ``ana1Tests/energyLineTestCfg.ini``.
+
+Additionally, there is the option to (re-)write ``local_moduleNameCfg.ini`` configuration files for all modules
+where a corresponding *collectionName_moduleName_override* section is provided by using
+the function: :py:func:`in3Utils.cfgHandling.rewriteLocalCfgs`. An example of this functionality is provided in
+:py:mod:`runScripts.runPlotContoursFromAsc.py` with the respective ``runPlotContoursFromAscCfg.ini`` file.
+
+.. Note::
+   By using :py:func:`in3Utils.cfgHandling.rewriteLocalCfgs`, already existing ``local_moduleNameCfg.ini`` files will be
+   overwritten for the respective modules.
```
Comment (fso42) [inline docs/configuration.rst:71]: Since this is not how it works at the moment, we could / should take this out, right?

**Review by fso42 — CHANGES_REQUESTED** [review]

**Review by fso42 — APPROVED** [review]

**Comment by fso42** [comment]
Standardtests ok, apart from avaKot

**Comment by fso42** [comment]
avaKot is related to change here: #948, accepting for now (new DEM data basis, reporjection ...). Benchmark results need to be updated.

**Pull request merged**